discovery/gossiper.go

```diff
@@ -1787,9 +1803,37 @@ func (d *AuthenticatedGossiper) processNetworkAnnouncement(
 		}
 
 		// Before we perform any of the expensive checks below, we'll
-		// make sure that the router doesn't already have a fresher
-		// announcement for this edge.
+		// make sure that we don't process any messages for known zombie
+		// edges.
 		timestamp := time.Unix(int64(msg.Timestamp), 0)
+		if d.cfg.Router.IsZombieEdge(msg.ShortChannelID) {
+			// If this is considered a zombie edge, we'll look at
+			// the current update's timestamp to determine if
+			// processing it would deem the edge as live. If it
+			// doesn't, we'll go ahead and reject the edge.
+			if time.Since(timestamp) > d.cfg.LiveEdgeHorizon {
+				nMsg.err <- fmt.Errorf("rejecting "+
+					"ChannelUpdate(%v) for zombie channel",
+					msg.ShortChannelID)
+				return nil
+			}
+
+			// Otherwise, we'll mark the edge as live within the
+			// router to ensure we can process any future updates
+			// for this edge until it is considered zombie again.
+			err := d.cfg.Router.MarkEdgeLive(msg.ShortChannelID)
```

Comment:
we should defer marking an edge as live only after we are sure  the signature validates, otherwise someone could blast updates w/ fake signatures that bring about a zombie uprising

Reply:
Since the `GetChannelByID` call below will fail due to it being a zombie channel, we'll go ahead and stash it so that it gets reprocessed once we receive its corresponding channel announcement. Without the announcement, we're unable to verify signatures since we don't have the public keys of each edge. If and when we receive the announcement, we won't be able to re-process it because it'll still be marked as a zombie.

Reply:
Fixed!